pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -2427,14 +2427,8 @@ public void onUpdate(TopicPolicies policies) {
         }
     }
 
-    private Optional<Policies> getNamespacePolicies(){
-        try {
-            return Optional.ofNullable(brokerService.pulsar().getAdminClient().namespaces()
-                    .getPolicies(TopicName.get(topic).getNamespace()));
-        } catch (Exception e) {
-            log.error("get namespace policies fail", e);
-        }
-        return Optional.empty();
+    private Optional<Policies> getNamespacePolicies() {
+        return DispatchRateLimiter.getPolicies(brokerService, topic);
```

Comment:
Why do we need to get the namespace policy from the DispatchRateLimiter? If we don't enable the dispatch rate limiter, is it possible to get the namespace policy?

Reply:
This is a static tool method, and has nothing to do with whether DispatchRateLimiter is enabled

Reply:
Ok